x-pack/plugins/snapshot_restore/__jest__/client_integration/snapshot_list.test.tsx

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { EuiSearchBoxProps } from '@elastic/eui/src/components/search_bar/search_box';
+
+import { useLoadSnapshots } from '../../public/application/services/http';
+import { DEFAULT_SNAPSHOT_LIST_PARAMS } from '../../public/application/lib';
+
+import * as fixtures from '../../test/fixtures';
+import { SnapshotListTestBed } from './helpers/snapshot_list.helpers';
+import { REPOSITORY_NAME } from './helpers/constant';
+import { pageHelpers, getRandomString } from './helpers';
+
+/*
+ * We are mocking useLoadSnapshots instead of sinon fake server because it's not
+ * spying on url parameters used in requests, for example /api/snapshot_restore/snapshots
+ * ?sortField=startTimeInMillis&sortDirection=desc&pageIndex=0&pageSize=20
+ * &searchField=repository&searchValue=test&searchMatch=must&searchOperator=exact
+ * would be shown as url=/api/snapshot_restore/snapshots is sinon server
+ */
+jest.mock('../../public/application/services/http', () => ({
+  useLoadSnapshots: jest.fn(),
+  setUiMetricServiceSnapshot: () => {},
+  setUiMetricService: () => {},
+}));
+
+/*
+ * Mocking EuiSearchBar because its onChange is not firing during tests
+ */
+jest.mock('@elastic/eui/lib/components/search_bar/search_box', () => {
+  return {
+    EuiSearchBox: (props: EuiSearchBoxProps) => (
+      <input
+        data-test-subj={props['data-test-subj'] || 'mockSearchBox'}
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+          props.onSearch(event.target.value);
+        }}
+      />
+    ),
+  };
+});
+
+const { setup } = pageHelpers.snapshotList;
+
+describe('<SnapshotList />', () => {
+  let testBed: SnapshotListTestBed;
+  let setSearchText: SnapshotListTestBed['actions']['setSearchText'];
+  let searchErrorExists: SnapshotListTestBed['actions']['searchErrorExists'];
+  let getSearchErrorText: SnapshotListTestBed['actions']['getSearchErrorText'];
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+    const snapshot = fixtures.getSnapshot({
+      repository: REPOSITORY_NAME,
+      snapshot: `a${getRandomString()}`,
```

Comment:
is the `a` prefix on purpose?

Reply:
Good catch! I copy-pasted this from a different test, not needed here.